test/controllers/api/v2/subscriptions_controller_test.rb

```diff
@@ -63,6 +64,15 @@ def test_index_protected
       end
     end
 
+    def test_index_with_activation_key_id
+      activation_key = ActivationKey.new
+      Katello::ActivationKey.expects(:find_by).returns(activation_key)
+      Katello::Pool::ActiveRecord_Relation.any_instance.expects(:for_activation_key).returns(Pool.readable)
+      get :index, activation_key_id: 1
+      assert_response :success
+      assert_template 'api/v2/subscriptions/index'
```

Comment:
I'd like to trust the model to behave properly and not test it in the controller. In this case, we're just making sure we're using our `for_activation_key` method, but the test for that is in the `Pool` model tests.